### DISCUSSION TITLE
How does the Graphcache work?

### DISCUSSION THREAD

**Discussion by stingalleman** [urql-graphql/urql#1822] [Q&A] [answered]
Hi,

I'm a bit confused about the Graphcache. Does it automatically cache, or not do I need to specify it all manually? I've read the docs but it's still a bit confusing.

**Comment by JoviDeCroock** [comment, marked as answer]
It's a cache just like the other one but the difference is how it caches, as the explanation details, graphCache is a [normalized cache](https://formidable.com/open-source/urql/docs/graphcache/) and the default is a [document cache](https://formidable.com/open-source/urql/docs/basics/document-caching/).

The normalized cache will gather query-results and cache appropriately but sometimes when we're mutating data you'll need to help it out a bit. When an entity is present in the cache and we update it the normalized cache will be able to reconcile that against the data it has.

```js
{ id: 1, text: 'hi', __typename: 'Todo' }
```

When the cache sees a response like this it's easy to for instance update the `Todo:1.text` to `hi` rather than what it was before, however when we are deleting/adding entities we'll need to help the cache a bit by doing:

```
Mutation: { deleteTodo: yourResolver }
```

The manual work only happens on select mutations and you'll notice when this happens because you'll see stale data.

**Reply by stingalleman** [reply]
Thank you, it is much more clearly now. So, with the normalized cache, I need to specify all mutations manually, while this isn't the case with the document cache, correct?

**Reply by JoviDeCroock** [reply]
No, you only need to specify mutations/subscriptions if the update is hard to reconcile like creating or deleting a new entity https://formidable.com/open-source/urql/docs/graphcache/cache-updates/ goes more into detail

**Reply by stingalleman** [reply]
Alright, awesome. Thank you so much for replying so fast and with so much detail and thank you to the entire team for making urql ❤️
